src/pages/on-chain-governance/shared/getCycleEndTime.tsx

```diff
@@ -0,0 +1,35 @@
+import { format, fromUnixTime } from "date-fns";
+
+export function getCycleEndTime(
+  cycleEndHeight: number,
+  currentBlockHeight: number,
+  currentBlockMedianTime: number,
+  connection: string
+) {
+  const timeDifferenceInBlocks = cycleEndHeight - currentBlockHeight;
+  let blockSeconds = 30;
+  switch (connection) {
+    case "Playground":
+      blockSeconds = 3;
+      break;
+    case "TestNet":
+      blockSeconds = 3;
+      break;
+    case "MainNet":
+    default:
+      blockSeconds = 30;
+  }
+
+  let cycleEndMedianTime = 0;
+  if (timeDifferenceInBlocks < 0) {
+    cycleEndMedianTime =
+      currentBlockMedianTime - Math.abs(timeDifferenceInBlocks) * blockSeconds;
```

Comment:
do we need this? can we replace this with line 29?

Reply:
this is to account for when currentBlockHeight > cycleEndHeight

Reply:
yea exactly, mathematically speaking, isn't it the same as line 29?

Reply:
ohh righttt updated my bad